### PR TITLE
Add tasks guidelines

### PR DESCRIPTION
**Pull request by N4ki**
When merged these changes will:
- title
- add english version blatantly copied from exisiting README
- fix obsolete task icons link and sqf formatting indicator (eng version)

### CODE REVIEW

File: docs/mods/armaforces_tasks.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        description = "Great mission of beating horse."; // Opis taska. Nie korzysta z łamaczy tekstu (klawisz enter). Jeżeli jest ci to potrzebne, użyj stringtable.
+        /* Ikona taska na mapie
+        Najpierw następuje sprawdzenie, czy znacznik/marker o danej nazwie istnieje. Jeżeli nie istnieje, to wyszukuje obiektu w namespace'ie misji.
+        Alternatywnym rozwiązaniem jest dostarczenie kooordynatów {x, y, z}.
+        Jeżeli nie pojawi się żaden z tych wpisów, to task będzie na mapie niewidoczny.*/
+        position[] = {}; // Do wykorzystania przy podawaniu konkretnych koordynatów
+        object = ""; // Do wykorzystanie dla nazw obiektów
+        marker = ""; // Do wykorzystania dla nazw znaczników/markerów
+        icon = "unknown"; // jak ikonka ma wyglądać na mapie - https://community.bistudio.com/wiki/Arma_3_Task_Framework#Task_icons - tutaj znajdziecie wylistowane możliwości.
+        owners[] = { "true" }; // Domyślna wartość, wpisz "All"/"true", aby task był dostępny dla wszystkich grających.
```
Comment (3Mydlo3) [inline docs/mods/armaforces_tasks.md:29]:
```suggestion
        owners[] = { "true" }; // Domyślna wartość "All"/"true" pokazuje zadanie wszystkim graczom.
```

File: docs/mods/armaforces_tasks.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+class CfgTasks {
+    class Soapy_Mission_XD {
+        title = "Soapy Mission XD - Horse Knocked"; // Nazwa taska
+        description = "Great mission of beating horse."; // Opis taska. Nie korzysta z łamaczy tekstu (klawisz enter). Jeżeli jest ci to potrzebne, użyj stringtable.
+        /* Ikona taska na mapie
+        Najpierw następuje sprawdzenie, czy znacznik/marker o danej nazwie istnieje. Jeżeli nie istnieje, to wyszukuje obiektu w namespace'ie misji.
+        Alternatywnym rozwiązaniem jest dostarczenie kooordynatów {x, y, z}.
+        Jeżeli nie pojawi się żaden z tych wpisów, to task będzie na mapie niewidoczny.*/
+        position[] = {}; // Do wykorzystania przy podawaniu konkretnych koordynatów
+        object = ""; // Do wykorzystanie dla nazw obiektów
```
Comment (3Mydlo3) [inline docs/mods/armaforces_tasks.md:26]:
```suggestion
        object = ""; // Do wykorzystanie dla nazwy obiektu
```

File: docs/mods/armaforces_tasks.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        marker = ""; // Do wykorzystania dla nazw znaczników/markerów
```
Comment (3Mydlo3) [inline docs/mods/armaforces_tasks.md:27]:
```suggestion
        marker = ""; // Do wykorzystania dla nazwy znacznika/markera
```

File: docs/mods/armaforces_tasks.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        icon = "unknown"; // jak ikonka ma wyglądać na mapie - https://community.bistudio.com/wiki/Arma_3_Task_Framework#Task_icons - tutaj znajdziecie wylistowane możliwości.
+        owners[] = { "true" }; // Domyślna wartość, wpisz "All"/"true", aby task był dostępny dla wszystkich grających.
+        initialState = "CREATED"; // Domyślna wartość
+        priority = -1; // Domyślna wartość
+        createdShowNotification = "false"; // Domyślna wartość
+        visibleIn3D = "false"; // Domyślna wartość
+        parentTask = ""; // Do konfigurowania nazwy taska nadrzędnego
```
Comment (3Mydlo3) [inline docs/mods/armaforces_tasks.md:34]:
```suggestion
        parentTask = ""; // Umożliwia wskazanie zadania nadrzędnego
```

File: docs/mods/armaforces_tasks.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        conditionCodeFailed = ""; // Task spartaczony
+        conditionCodeCanceled = ""; // Task anulowany
+        // Kody eventów CBA, które muszą być odpalone, żeby  zakończyć task
+        conditionEventsSuccess[] = {}; // Task udany
+        conditionEventsFailed[] = {}; // Task spartaczony
+        conditionEventsCanceled[] = {}; // Task anulowany
+        // Ile eventów musi zakończyć się sukcesem/porażką/anulowaniem, aby dany task się uruchomił
+        conditionEventsSuccessRequired = 1;  // sukces
+        conditionEventsFailedRequired = 1; // porażka
+        conditionEventsCanceledRequired = 1; // anulowanie
```
Comment (3Mydlo3) [inline docs/mods/armaforces_tasks.md:52]:
```suggestion
        conditionEventsSuccessRequired = 1;  // Task udany
        conditionEventsFailedRequired = 1; // Task spartaczony
        conditionEventsCanceledRequired = 1; // Task anulowany
```

File: docs/mods/armaforces_tasks.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+        // Kody warunków, które musza zwracać true, aby zakończyć task
+        conditionCodeSuccess = ""; // Task udany
+        conditionCodeFailed = ""; // Task spartaczony
+        conditionCodeCanceled = ""; // Task anulowany
+        // Kody eventów CBA, które muszą być odpalone, żeby  zakończyć task
+        conditionEventsSuccess[] = {}; // Task udany
+        conditionEventsFailed[] = {}; // Task spartaczony
+        conditionEventsCanceled[] = {}; // Task anulowany
+        // Ile eventów musi zakończyć się sukcesem/porażką/anulowaniem, aby dany task się uruchomił
```
Comment (3Mydlo3) [inline docs/mods/armaforces_tasks.md:49]:
```suggestion
        // Ile eventów z odpowiadającej listy musi zostać wywołanych aby zadanie zostało zakończone w określony sposób
```

File: docs/mods/armaforces_tasks.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+        // Wywoływanie eventów CBA. Wywoływanie customowych zdarzeń i kodu odbywa się przez dodanie odpowiednich event handlerów CBA na serwerze - patrz niżej.
+        onShowEvents[] = {}; // eventy wywoływane przy ujawnieniu taska
+        onSuccessEvents[] = {}; // eventy wywoływane przy sukcesie taska
+        onFailedEvents[] = {}; // eventy wywoływane przy spartaczeniu, porażce taska
+        onCanceledEvents[] = {}; // eventy wywoływane przy anulowaniu taska
+    };
+
+    // definiujemy taski podrzędne
+    class FindHorse { // nazwa klasy
+        title = "Znajdź kunia"; // tytuł, nazwa taska
```
Comment (3Mydlo3) [inline docs/mods/armaforces_tasks.md:63]:
```suggestion
        title = "Znajdź kunia"; // Tytuł taska
```

File: docs/mods/armaforces_tasks.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+        // Wywoływanie eventów CBA. Wywoływanie customowych zdarzeń i kodu odbywa się przez dodanie odpowiednich event handlerów CBA na serwerze - patrz niżej.
+        onShowEvents[] = {}; // eventy wywoływane przy ujawnieniu taska
+        onSuccessEvents[] = {}; // eventy wywoływane przy sukcesie taska
+        onFailedEvents[] = {}; // eventy wywoływane przy spartaczeniu, porażce taska
+        onCanceledEvents[] = {}; // eventy wywoływane przy anulowaniu taska
+    };
+
+    // definiujemy taski podrzędne
+    class FindHorse { // nazwa klasy
```
Comment (3Mydlo3) [inline docs/mods/armaforces_tasks.md:62]:
```suggestion
    class FindHorse { // nazwa klasy taska
```

File: docs/mods/armaforces_tasks.md [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+        title = "Znajdź kunia"; // tytuł, nazwa taska
+        icon = "search"; // ikona, którą zobaczą na tasku gracze
+        parentTask = "Soapy_Mission_XD"; // Task nadrzędny, odwołanie następuje po nazwie klasy
+        conditionCodeShow = "true"; // Task będzie widoczny dla graczy na liście tasków
```
Comment (3Mydlo3) [inline docs/mods/armaforces_tasks.md:66]:
```suggestion
        conditionCodeShow = "true"; // Task będzie od razu widoczny dla graczy na liście tasków
```

File: docs/mods/armaforces_tasks.md [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+        conditionCodeSuccess = "unit1 distance horse < 50"; // Sukces taska nastąpi w momencie, kiedy dystans między obiektami unit1 oraz horse < 50
+        onSuccessEvents[] = { "horseFound" }; // event wywoływany zaliczeniem taska (odwołanie nastąpi nieco niżej, czytajcie uważnie)
+    };
+    class KnockHorse { 
+        title = "Zwal kunia"; 
+        icon = "attack"; 
+        object = "horse"; // Wskazuje nazwę konkretnego obiektu, który będzie posiadał na sobie powyższą ikonę
```
Comment (3Mydlo3) [inline docs/mods/armaforces_tasks.md:73]:
```suggestion
        object = "horse"; // Wskazuje nazwę konkretnego obiektu, za którym będzie podążał task na mapie
```

File: docs/mods/armaforces_tasks.md [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+        parentTask = "FindHorse";
+
+        conditionEventsShow[] = { "horseFound" }; // warunek, który musi być spełniony, by task się pojawił. Patrz linia 66
```
Comment (3Mydlo3) [inline docs/mods/armaforces_tasks.md:76]:
```suggestion
        conditionEventsShow[] = { "horseFound" }; // warunek, który musi być spełniony, by task się pojawił. Patrz onSuccessEvents[] dla zadania FindHorse
```

File: docs/mods/armaforces_tasks.md [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+        conditionCodeSuccess = "!(alive horse)"; // by zaszło zwycięstwo obiekt "horse" ma być martwy (negacją jest znak "!")
+        conditionCodeFailed = "!(alive unit1)"; // by zaszła porażka obiekt "unit1" ma być martwy
+
+        onSuccessEvents[] = { "horseKnocked" }; // event wywołany udanym zakończeniem taska, patrz niżej
+        onFailedEvents[] = { "playerDied" }; // event wywołany porażką taska, patrz niżej
```
Comment (3Mydlo3) [inline docs/mods/armaforces_tasks.md:81]:
```suggestion
        onSuccessEvents[] = { "horseKnocked" }; // event wywołany udanym zakończeniem taska, przykład obsługi w rozdziale "Przykładowy niestandardowy event handling"
        onFailedEvents[] = { "playerDied" }; // event wywołany porażką taska, przykład obsługi w rozdziale "Przykładowy niestandardowy event handling"
```

File: docs/mods/armaforces_tasks.md [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+["playerDied", { // nazwa eventu wywoływana w description.ext
+    titleText ["You died!", "PLAIN DOWN", 0.5];
+    [{
+        "EveryoneLost" call BIS_fnc_endMission;
+    }, [], 3] call CBA_fnc_waitAndExecute;
+}] call CBA_fnc_addEventHandler;
+```
+
+## Framework events
+
```
Comment (3Mydlo3) [inline docs/mods/armaforces_tasks.md:109]:
```suggestion

Zaawansowana obsługa eventów CBA wywoływanych przez framework tasków. Dostępne są 2 eventy:

- `afm_tasks_taskCreated` wywoływany jest gdy zadanie zostaje utworzone (czyli w momencie spełnienia warunku utworzenia zadania). Jako parametr posiada nazwę klasy zadania, które zostało utworzone.
- `afm_tasks_taskStateChanged` wywoływany jest, gdy zadanie zmieni swój stan. Jako parametry posiada nazwę klasy zadania oraz nowy stan zadania.

```

**Review by 3Mydlo3 — CHANGES_REQUESTED** [review]
Good, just some small enhancements should be done. 🙂

If you don't agree with some proposed changes, feel free to adjust them yourself. 😛

**Comment by N4ki** [comment]
Much obliged, a fair amount of enhancements and "typo" fixes :)

**Review by 3Mydlo3 — APPROVED** [review]
Nicely done wiki page. 🙂 Hope our mission makers will find it useful.

**Pull request merged**